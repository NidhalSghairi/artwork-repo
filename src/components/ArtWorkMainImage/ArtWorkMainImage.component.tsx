import React from "react";
import { Grid } from "@mui/material";
import RemoveRedEyeIcon from "@mui/icons-material/RemoveRedEye";
import ViewInArIcon from "@mui/icons-material/ViewInAr";
import { StyledArtworkImage } from "./ArtWorkMainImage.style";

type Props = {
  imageUrl: string;
};
const ArtWorkMainImage: React.FC<Props> = ({ imageUrl }: Props) => {
  return (
    <Grid container>
      <Grid item xs={12}>
        <div>
          <StyledArtworkImage src={imageUrl} />
        </div>
      </Grid>
      <Grid
        item
        container
        xs={12}
        display="flex"
        justifyContent="center"
        columnSpacing={4}
      >
        <Grid item display="flex" alignItems="center">
          <RemoveRedEyeIcon style={{ marginRight: "8px" }} /> view in a room
        </Grid>
        <Grid item display="flex" alignItems="center">
          <ViewInArIcon style={{ marginRight: "8px" }} /> AR view
        </Grid>
      </Grid>
    </Grid>
  );
};

export default ArtWorkMainImage;
