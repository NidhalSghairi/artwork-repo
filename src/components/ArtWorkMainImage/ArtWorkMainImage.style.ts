import styled from "styled-components";

export const StyledArtworkImage = styled.img`
  width: 100%;
  height: 500px;
`;
