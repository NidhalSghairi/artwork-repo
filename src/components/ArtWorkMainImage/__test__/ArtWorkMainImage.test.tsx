import React from "react";
import renderer from "react-test-renderer";
import ArtWorkMainImage from "../ArtWorkMainImage.component";

it("should renders correctly", () => {
  const props = {
    imageUrl:
      "https://images.saatchiart.com/saatchi/655989/art/8520384/7584428-BHHQLFRA-7.jpg",
  };
  const tree = renderer.create(<ArtWorkMainImage {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
