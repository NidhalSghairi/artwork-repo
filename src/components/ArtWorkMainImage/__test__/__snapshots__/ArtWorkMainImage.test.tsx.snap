// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <div>
      <img
        className="sc-bcXHqe dfuyyN"
        src="https://images.saatchiart.com/saatchi/655989/art/8520384/7584428-BHHQLFRA-7.jpg"
      />
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-1gnp0z6-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-ev762u-MuiGrid-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="RemoveRedEyeIcon"
        focusable="false"
        style={
          Object {
            "marginRight": "8px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
        />
      </svg>
       view in a room
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-ev762u-MuiGrid-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ViewInArIcon"
        focusable="false"
        style={
          Object {
            "marginRight": "8px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="m18.25 7.6-5.5-3.18c-.46-.27-1.04-.27-1.5 0L5.75 7.6c-.46.27-.75.76-.75 1.3v6.35c0 .54.29 1.03.75 1.3l5.5 3.18c.46.27 1.04.27 1.5 0l5.5-3.18c.46-.27.75-.76.75-1.3V8.9c0-.54-.29-1.03-.75-1.3zM7 14.96v-4.62l4 2.32v4.61l-4-2.31zm5-4.03L8 8.61l4-2.31 4 2.31-4 2.32zm1 6.34v-4.61l4-2.32v4.62l-4 2.31zM7 2H3.5C2.67 2 2 2.67 2 3.5V7h2V4h3V2zm10 0h3.5c.83 0 1.5.67 1.5 1.5V7h-2V4h-3V2zM7 22H3.5c-.83 0-1.5-.67-1.5-1.5V17h2v3h3v2zm10 0h3.5c.83 0 1.5-.67 1.5-1.5V17h-2v3h-3v2z"
        />
      </svg>
       AR view
    </div>
  </div>
</div>
`;
